app/routers/event.py

```diff
@@ -72,32 +82,100 @@ async def create_new_event(request: Request, session=Depends(get_db)):
     if is_zoom:
         raise_if_zoom_link_invalid(location)
 
-    event = create_event(session, title, start, end, owner_id, content,
-                         location, invitees=invited_emails,
+    event = create_event(session, title, start, end, owner_id, privacy,
+                         content, location, invitees=invited_emails,
                          category_id=category_id,
                          availability=availability)
 
     messages = get_messages(session, event, uninvited_contacts)
     return RedirectResponse(router.url_path_for('eventview', event_id=event.id)
-                            + f'messages={"---".join(messages)}',
+                            + f'?messages={"---".join(messages)}',
                             status_code=status.HTTP_302_FOUND)
 
 
+def nonexisting_event(event_id: int) -> None:
+    error_message = f"Event ID does not exist. ID: {event_id}"
+    logger.exception(error_message)
+    raise HTTPException(
+        status_code=status.HTTP_404_NOT_FOUND,
+        detail=error_message)
+
+
 @router.get("/{event_id}")
 async def eventview(request: Request, event_id: int,
                     db: Session = Depends(get_db)):
     event = by_id(db, event_id)
+    event_to_show = can_show_event(event, db)
+    if not event_to_show:
+        nonexisting_event(event.id)
     start_format = '%A, %d/%m/%Y %H:%M'
     end_format = ('%H:%M' if event.start.date() == event.end.date()
                   else start_format)
     messages = request.query_params.get('messages', '').split("---")
     return templates.TemplateResponse("event/eventview.html",
-                                      {"request": request, "event": event,
+                                      {"request": request,
+                                       "event": event_to_show,
                                        "start_format": start_format,
                                        "end_format": end_format,
                                        "messages": messages})
 
 
+class PrivateEvent:
```

Comment:
Maybe we should put this in `internal/privacy`?

Reply:
Done :)